tekton/build_pipeline.go

```diff
@@ -114,3 +123,9 @@ func GenerateSHA(str string) string {
 	hash := sha256.Sum256([]byte(str))
 	return fmt.Sprintf("%x", hash)[0:62]
 }
+
+// IsPLRCreatedByPACPullRequestEvent checks if a PLR has label PipelineAsCodeEventTypeLabel and with pull_request or merge request value
+func IsPLRCreatedByPACPullRequestEvent(plr *tektonv1.PipelineRun) bool {
+	return metadata.HasLabelWithValue(plr, PipelineAsCodeEventTypeLabel, PipelineAsCodePullRequestType) ||
+		metadata.HasLabelWithValue(plr, PipelineAsCodeEventTypeLabel, PipelineAsCodeMergeRequestType)
```

Comment:
I think that the issue here might be for retest events (when users comment `/retest` on their PR), because that event type is (annoyingly) different from the initial `pull_request` type: 
``` 
pipelinesascode.tekton.dev/event-type: retest-all-comment 
```
Since other types of comments might trigger rebuilds or retests, we might need to consider doing a similar approach as for Snapshots with the [IsSnapshotCreatedByPACPushEvent](https://github.com/konflux-ci/integration-service/blob/a0b2ceb838ee73f33447a5cfca34e2bc4b7955b7/gitops/snapshot.go#L607-L613) function.

Reply:
Thanks! I forget comment can also trigger the pipelineRun. It is impressive now. Updated.